pkg/buildcontext/azureblob.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package buildcontext
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/GoogleContainerTools/kaniko/pkg/constants"
+	"github.com/GoogleContainerTools/kaniko/pkg/util"
+)
+
+// AzureBlob struct for Azure Blob Storage processing
+type AzureBlob struct {
+	context string
+}
+
+// Download context file from given azure blob storage url and unpack it to BuildContextDir
+func (b *AzureBlob) UnpackTarFromBuildContext() (string, error) {
+
+	//Get Azure_STORAGE_ACCESS_KEY from environment variables
+	accountKey := os.Getenv("AZURE_STORAGE_ACCESS_KEY")
+	if len(accountKey) == 0 {
+		return "", errors.New("AZURE_STORAGE_ACCESS_KEY environment variable is not set")
+	}
+
+	//Get storage accoutname for Azure Blob Storage
+	u, _ := url.Parse(b.context)
+	parts := azblob.NewBlobURLParts(*u)
+	accountName := strings.Split(parts.Host, ".")[0]
+
+	//Generate credentail with accountname and accountkey
+	credential, err := azblob.NewSharedKeyCredential(accountName, accountKey)
+	if err != nil {
+		return parts.Host, err
+	}
+
+	//Create directory and file for downloading context file
+	directory := constants.BuildContextDir
+	tarPath := filepath.Join(directory, constants.ContextTar)
+	if err := os.MkdirAll(directory, 0750); err != nil {
+		return directory, err
+	}
+	file, err := os.Create(tarPath)
```

Comment:
i think you can re-use util.CreateFile https://github.com/GoogleContainerTools/kaniko/blob/master/pkg/buildcontext/gcs.go#L71
```
if err := util.CreateFile(tarPath, reader, 0600, 0, 0); err != nil {
		return "", err
	}
```

Reply:
Also, looks like this piece of code is repeated even in [s3.go](https://github.com/GoogleContainerTools/kaniko/blob/8acab90decb3e75fe61de8dea0cdb8bd8c11cc78/pkg/buildcontext/s3.go#L59)
```
	directory := constants.BuildContextDir
	tarPath := filepath.Join(directory, constants.ContextTar)
	if err := os.MkdirAll(directory, 0750); err != nil {
		return directory, err
	}
	file, err := os.Create(tarPath)
	if err != nil {
		return directory, err
	}
```

Would you mind creating a util like
`util.setUpBuildContextAndTarpath`

Reply:
Thanks @tejal29  for review and suggestion. I will make this improvement shortly. thanks!

Reply:
thank you for your contribution! Looking forward to get this in!!

Reply:
@tejal29 I have pushed some changes to reflect our discussion 

> i think you can re-use util.CreateFile https://github.com/GoogleContainerTools/kaniko/blob/master/pkg/buildcontext/gcs.go#L71
> 
> ```
> if err := util.CreateFile(tarPath, reader, 0600, 0, 0); err != nil {
> 		return "", err
> 	}
> ```
The CreateFile in util do not only create the file but also copy the content to the file through a reader. I intended to let the sdk to handle the file directly. I wrapped the make directory and create file part into a new function CreateTargetTarfile and reused in azureblob.go. Let me know how it looks for you.

Thanks!